src/encoded/static/components/cart/lock.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { svgIcon } from '../../libs/svg-icons';
+import { setCartLockAndSave } from './actions';
+
+
+/**
+ * Renders a lock icon which, on a user click, initiates the locking of the given cart. That
+ * involves setting the `locked` flag in the cart object, though all other processing of this flag
+ * happens in the front end.
+ */
+const CartLockTriggerComponent = ({ savedCartObj, inProgress, onLock }) => {
+    // Called when the user clicks the lock/unlock button.
+    const handleLockClick = () => {
+        onLock(!savedCartObj.locked);
+    };
+
+    // Determine the tooltip text.
+    let disabledTooltip = '';
+    if (savedCartObj.status === 'deleted') {
+        disabledTooltip = 'Cannot lock a deleted cart';
+    } else if (savedCartObj.status === 'disabled') {
+        disabledTooltip = 'Cannot share the auto-save cart';
+    } else if (inProgress) {
+        disabledTooltip = 'Cart operation in progress';
+    }
+
+    return (
+        <div className="cart-manager-table__tooltip-group">
+            {disabledTooltip ?
+                <div
+                    className="cart-manager-table__button-overlay"
+                    title={disabledTooltip}
+                />
+            : null}
+            <button
+                onClick={handleLockClick}
+                className="btn btn-sm btn-warning btn-inline"
+                disabled={inProgress || savedCartObj.status === 'deleted' || savedCartObj.status === 'disabled'}
+                aria-label={`${savedCartObj.locked ? 'Unlock' : 'Lock'} cart`}
+            >
+                {svgIcon(savedCartObj.locked ? 'lockClosed' : 'lockOpen', { width: 13, marginRight: 4 })}
+                {savedCartObj.locked ? <span>Unlock</span> : <span>Lock</span>}
+            </button>
```

Comment:
Another really minor comment - did you think about making the Lock and Unlock buttons the same size? The other buttons have the same names so their buttons are automatically the same width, and you could do that if you wanted with the Lock/Unlock buttons. Really not a big deal, just a thought as I was looking at these buttons.
![Screen Shot 2020-06-08 at 5 25 16 PM](https://user-images.githubusercontent.com/2887343/84092758-3614f180-a9ad-11ea-80ac-23b1782eb76d.png)

Reply:
Fixed. I tried to do this in a way that the button is only as wide as the widest possible label, and wrote a mostly-working mechanism to do it, but it was so complicated that it hardly seemed worth it when I can just hard code one line of CSS to do what’s probably good enough, so that’s what I went with.